iso20022-core/src/main/java/com/prowidesoftware/swift/model/mx/MxParseUtils.java

```diff
@@ -241,4 +247,174 @@ private static Optional<MxId> enrichBusinessService(MxId mxId, final String xml)
+     * For example, to find the {@code <Cd>} tag within {@code <ClrSys>}, you would call:
+     * <pre>
+     *     findElement(xml, "ClrSys", "Cd");
+     * </pre>
+     *
+     * @param xml  the XML document as a {@link String} to search.
+     * @param tags the sequence of tag names that define the hierarchy of the target element.
+     * @return an {@link Optional} containing the {@link XMLStreamReader} positioned at the
+     * matching element if found; otherwise, an empty {@link Optional}.
+     * @throws NullPointerException     if the {@code xml} or {@code tags} are null.
+     * @throws IllegalArgumentException if the {@code xml} is a blank string.
+     * @since 9.5.5
+     */
+    public static Optional<XMLStreamReader> findElementByTags(final String xml, String... tags) {
+        Objects.requireNonNull(xml, "XML to parse must not be null");
+        Validate.notBlank(xml, "XML to parse must not be a blank string");
+        Objects.requireNonNull(xml, "tags to find must not be null");
+
+        final XMLInputFactory xif = SafeXmlUtils.inputFactory();
+        int tagsIndex = 0;
+        try {
+            final XMLStreamReader reader =
+                    xif.createXMLStreamReader(new StringReader(MxParseUtils.makeXmlLenient(xml)));
+            while (reader.hasNext()) {
+                int event = reader.next();
+                if (XMLStreamConstants.START_ELEMENT == event) {
+                    if (reader.getLocalName().equals(tags[tagsIndex])) {
+                        if (tagsIndex == tags.length - 1) {
+                            return Optional.of(reader);
+                        }
+                        tagsIndex++;
+                    }
+                }
+            }
+        } catch (XMLStreamException e) {
+            log.log(Level.WARNING, "Error reading XML", e);
+        }
+        return Optional.empty();
+    }
+
+    /**
+     * Finds an XML element within a document by traversing a specified tag hierarchy.
+     *
+     * <p>This method uses an {@link XMLStreamReader} to parse the provided XML document.
+     * It searches for an element that matches the specified sequence of tag names (hierarchy).
+     * For example, to find the {@code <Cd>} tag within {@code <ClrSys>}, you would call:
+     * <pre>
+     *     findElement(xml, "ClrSys", "Cd");
+     * </pre>
+     *
+     * @param xml        the XML document as a {@link String} to search.
+     * @param targetPath the path of the field to find into the xml.
+     * @return an {@link Optional} containing the {@link XMLStreamReader} positioned at the
+     * matching element if found; otherwise, an empty {@link Optional}.
+     * @throws NullPointerException     if the {@code xml} or {@code tags} are null.
+     * @throws IllegalArgumentException if the {@code xml} is a blank string.
+     * @since 9.5.5
+     */
+    public static Optional<XMLStreamReader> findElementByAbsolutePath(String xml, String targetPath) {
```

Comment:
should be byPath and the relative path indicated with double slash at the beginning 

/A/B/C -> Absolute
//C/D -> relative
C/D -> same as absolute because you search C from the root